apollo-router/src/plugins/connectors/handle_responses.rs

```diff
@@ -48,116 +50,137 @@ pub(crate) async fn handle_responses(
     let count = responses.len();
 
     for response in responses {
-        let (parts, body) = response.into_parts();
-
-        let response_key = parts
-            .extensions
-            .get::<ResponseKey>()
-            .ok_or(MissingResponseKey)?;
-
-        let body = &hyper::body::to_bytes(body)
-            .await
-            .map_err(|_| InvalidResponseBody("couldn't retrieve http response body".into()))?;
-
-        if parts.status.is_success() {
-            let Ok(json_data) = serde_json::from_slice::<Value>(body) else {
-                if let Some(debug) = debug {
-                    debug.lock().push_invalid_response(&parts, body);
-                }
-                return Err(InvalidResponseBody(
-                    "couldn't deserialize response body".into(),
-                ));
-            };
-
-            let mut res_data = {
-                // TODO: caching of the transformed JSONSelection with the selection set applied?
-                let transformed_selection = connector
-                    .selection
-                    .apply_selection_set(response_key.selection_set());
-
-                let (res, apply_to_errors) = transformed_selection.apply_with_vars(
-                    &json_data,
-                    &response_key.inputs().merge(connector.config.as_ref()),
-                );
-
-                if let Some(ref debug) = debug {
-                    debug.lock().push_response(
-                        &parts,
-                        &json_data,
-                        Some(SelectionData {
-                            source: connector.selection.to_string(),
-                            transformed: transformed_selection.to_string(),
-                            result: res.clone(),
-                            errors: apply_to_errors,
-                        }),
-                    );
-                }
-                res.unwrap_or_else(|| Value::Null)
-            };
+        let mut error = None;
+        let response_key = response.1;
+        match response.0 {
+            Err(e) => error = Some((e.to_string(), e.extension_code())),
+            Ok(response) => {
+                let (parts, body) = response.into_parts();
+                let body = &hyper::body::to_bytes(body).await.map_err(|_| {
+                    InvalidResponseBody("couldn't retrieve http response body".into())
+                })?;
+
+                if parts.status.is_success() {
+                    let Ok(json_data) = serde_json::from_slice::<Value>(body) else {
+                        if let Some(debug) = debug {
+                            debug.lock().push_invalid_response(&parts, body);
+                        }
+                        return Err(InvalidResponseBody(
+                            "couldn't deserialize response body".into(),
+                        ));
+                    };
```

Comment:
```suggestion
                    let json_data = serde_json::from_slice::<Value>(body).map_err(|e| {
                        if let Some(debug) = debug {
                            debug.lock().push_invalid_response(&parts, body);
                        }
                        InvalidResponseBody(
                            format!("couldn't deserialize response body: {e}"),
                        )
                    })?;
```

Reply:
This is mostly to preserve the original error, but feel free to ignore.

Reply:
This code isn't really modified in this PR, only the indentation level changed, but I can include this.